credentials/src/main/kotlin/web5/sdk/credentials/ExceptionDeclarations.kt

```diff
@@ -41,3 +41,11 @@ public class StatusListCredentialFetchException(cause: Throwable, message: Strin
  */
 public class StatusListCredentialParseException(cause: Throwable, message: String? = null)
   : RuntimeException(message, cause)
+
+/**
+ * Presentation exchange validation exception.
+ *
+ * @param message the exception message detailing the error
+ */
+public class PexValidationException(message: String? = null)
```

Comment:
Thought on having the type of the exception be specific to the error? For example:

```kt
public class PresentationDefinitionIdEmptyException
```

I see that being the approach taken for the existing exceptions. One benefit is that clients can handle those exceptions by type programatically, instead of having to parse the message.

Reply:
yes I was thinking about this, but then we would have like a new exception for each different check basically. 15 different Exception classes seems excessive right?

Reply:
I'd say it depends. For example, did resolution has 10 different errors https://w3c-ccg.github.io/did-resolution/#errors

Perhaps some could be combined, so there's a balance in the middle. Up to you.